### PR TITLE
Add ethereum.networkVersion, removeAllListeners, and ethereum._metamask.isUnlocked for webcompat, wallet_addEthereumNetwork fix (Uplift for 1.33.x)

### PR DESCRIPTION
**Pull request by bbondy**
Improves webcompat with Dapps.
Also cherry-pick `Check keyring support in ShouldCreate1559Tx` because it is built upon one of the PRs uplifted here. 

Resolves https://github.com/brave/brave-browser/issues/19544
Resolves https://github.com/brave/brave-browser/issues/19551
Resolves https://github.com/brave/brave-browser/issues/19624
Resolves https://github.com/brave/brave-browser/issues/19617
Resolves https://github.com/brave/brave-browser/issues/19656
Resolves https://github.com/brave/brave-browser/issues/19291
Resolves https://github.com/brave/brave-browser/issues/19673

Uplift of: https://github.com/brave/brave-core/pull/11195
Uplift of: https://github.com/brave/brave-core/pull/11196
Uplift of: https://github.com/brave/brave-core/pull/11215
Uplift of: https://github.com/brave/brave-core/pull/11200
Uplift of: https://github.com/brave/brave-core/pull/11222
Uplift of: https://github.com/brave/brave-core/pull/11229

Pre-approval checklist: 
- [x] You have tested your change on Nightly. 
- [ ] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [x] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.34.31 Chromium: 96.0.4664.45 (Official Build) nightly (64-bit)
-- | --
Revision | `76e4c1bb2ab4671b8beba3444e61c0f17584b2fc-refs/branch-heads/4664@{#947}`
OS | Linux



**brave/brave-browser#19617**
- Ledger testing needs to be redone on a new build for EIP-1559 gas pricing.

Polygon | Ethereum | Ledger | Trezor
-- | -- | -- | --
<video src=https://user-images.githubusercontent.com/17010094/143019230-a677dcd9-e628-4a47-a993-63f440d841e9.mov> | <video src=https://user-images.githubusercontent.com/17010094/143019201-a1f6a558-9b00-4d96-a9eb-7747e51afa63.mov> | <video src=https://user-images.githubusercontent.com/17010094/143019217-622b1ca2-6d94-4df2-adcc-68db49355ce6.mov> | <video src=https://user-images.githubusercontent.com/17010094/143019241-fd3a1363-3343-413c-935b-1474fba781bc.mov>

**brave/brave-browser#19551**

https://user-images.githubusercontent.com/17010094/143018812-c4a41161-43c6-44ab-8412-78482ff85d29.mov


**brave/brave-browser#19624**

https://user-images.githubusercontent.com/17010094/143018833-64a87403-f413-4706-9f86-fb2126f06ddd.mov

**brave/brave-browser#19656**

https://user-images.githubusercontent.com/17010094/143018927-f110d07e-c499-4d89-8d6d-bd0216d89451.mov

**brave/brave-browser#19291**
- Verified able to switch network as per issue description

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.33.x` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-core/pull/11201#issuecomment-976610570.

**Pull request merged**
